crates/fluvio-cli/src/client/topic/add_partition.rs

```diff
@@ -0,0 +1,147 @@
+//!
+//! # Increment partition of a Topic
+//!
+//! CLI tree to increment the number of partitions of a topic.
+//!
+use clap::Parser;
+use anyhow::{anyhow, Result};
+use comfy_table::{Cell, Row, Table};
+use fluvio_types::ReplicaMap;
+use tokio::select;
+use futures::StreamExt;
+
+use fluvio_future::timer::sleep;
+use fluvio_sc_schema::topic::{AddPartition, TopicSpec, UpdateTopicAction};
+use fluvio::{Fluvio, FluvioAdmin};
+
+/// Option for Listing Partition
+#[derive(Debug, Parser)]
+pub struct AddPartitionOpt {
+    /// Topic name
+    topic: String,
+    /// Number of partitions to add
+    #[arg(short, long, default_value = "1")]
+    count: i32,
+}
+
+const CHECK_NEW_PARTITIONS_TIMEOUT_MS: u64 = 10000;
+
+impl AddPartitionOpt {
+    pub async fn process(self, fluvio: &Fluvio) -> Result<()> {
+        let admin = fluvio.admin().await;
+
+        let replica_map_before = admin
+            .list::<TopicSpec, _>(vec![self.topic.clone()])
+            .await?
+            .first()
+            .ok_or_else(|| anyhow!("Topic not found"))?
+            .status
+            .replica_map
+            .clone();
+
+        select! {
+            replica_map_after_result = self.send_new_partitions_and_watch_them(&replica_map_before, &admin) => {
+                match replica_map_after_result {
+                    Ok(replica_map_after) => {
+                        let diff = replica_map_after
+                            .into_iter()
+                            .filter(|(k, v)| {
+                                let before = replica_map_before.get(k);
+                                match before {
+                                    Some(before) => *before != *v,
+                                    None => true,
+                                }
+                            })
+                            .collect::<Vec<_>>();
+
+                        println!("added new partitions to topic: \"{}\"", self.topic);
+                        println!("{}", display_new_partitions_spu_table(diff));
+                    },
+                    Err(err) => {
+                        eprintln!("add partition failed: {err}");
+                    },
+                }
+            },
+            _ = sleep(std::time::Duration::from_millis(CHECK_NEW_PARTITIONS_TIMEOUT_MS)) => {
+                println!("response timeout exceeded");
+            },
+        }
+
+        Ok(())
+    }
+
+    async fn send_new_partitions_and_watch_them(
+        &self,
+        replica_map_before: &ReplicaMap,
+        admin: &FluvioAdmin,
+    ) -> Result<ReplicaMap> {
+        let request = AddPartition {
+            count: self.count as u32,
+        };
+
+        let action = UpdateTopicAction::AddPartition(request);
+        let mut partition_stream = admin.watch::<TopicSpec>().await?;
+        let mut connected = false;
+        loop {
+            if let Some(Ok(event)) = partition_stream.next().await {
```

Comment:
add timer so this doesn't result in infinite loop

Reply:
The timer outside does not cover it? 

https://github.com/infinyon/fluvio/pull/4065/files/48d0c616794f8f53f31c2ecf15fc41cddc6eb0be#diff-5660ec23133d43cf7faef763ee6f34d821c87a1ac45f4bb74f8a1e4154ae4f4cR65

Reply:
just move timer inside then can get rid of extras select in the outside

Reply:
Done, WDYT?